src/causalprog/graph/node.py

```diff
@@ -178,19 +178,20 @@ def sample(
         samples: int,
         rng_key: jax.Array,
     ) -> npt.NDArray[float]:
+        d = self._dist(
```

Comment:
Again this is more a note rather than something that needs to be done in this PR; but now that we're sticking to `numpyro` we can avoid implementing this sequential sampling down the nodes we've got going right now, and instead use `Graph.model` to generate the samples for us.